src/components/Table/DataTable/compenents/el-data-table/utils/select-strategy.js

```diff
@@ -177,19 +117,17 @@ class StrategyPersistSelection extends StrategyAbstract {
   updateElTableSelection() {
     const { data, id, selected } = this.elDataTable
 
-    if (!this.elTable) {
-      return
-    }
+    // 历史勾选的行已经不在当前页了，所以要将当前页的行数据和selected合并
+    const mergeData = _.uniqWith([...data, ...selected], _.isEqual)
 
-    // 先清空当前页选择状态
-    this.elTable.clearSelection()
+    mergeData.forEach(r => {
+      const isSelected = !!selected.find(r2 => r[id] === r2[id])
 
-    // 只处理当前页的数据
-    data.forEach(row => {
-      const isSelected = selected.some(r => r[id] === row[id])
-      if (isSelected) {
-        this.elTable.toggleRowSelection(row, true)
+      if (!this.elTable) {
+        return
       }
+
+      this.elTable.toggleRowSelection(r, isSelected)
     })
   }
 }
```

Comment:
There are no significant differences between the two code snippets provided. However, there might be some unnecessary operations present which can be removed:

```jsx
class StrategyPersistSelection extends StrategyAbstract {

  batchProcessSelection(rows) { ... }

  updateElTableSelection() { ...
}
```

Here's how you could refactor them slightly without altering their functionality significantly:
```typescript

export default class StrategyPersistSelection extends StrategyAbstract {
    // Other properties/Methods...
}

// Existing ES-6 imports to use classes and methods
import React from 'react';
...

const StrategyPersistSelection: React.FC = (_props) => {  
}
```